internal/clients/instance_description.go

```diff
@@ -0,0 +1,13 @@
+package clients
+
+// InstanceInfo defines a model for an instance description
+type InstanceDescription struct {
+	// The id of the instance
+	ID *string `json:"id,omitempty" yaml:"id"`
+
+	// The public ipv4 dns of the instance
+	PublicDNS *string `json:"dns,omitempty" yaml:"dns"`
+
+	// the public ipv4 of the instance
+	PublicIPv4 *string `json:"ipv4,omitempty" yaml:"ipv4"`
+}
```

Comment:
Can we not use string pointers? Pointers are bad for strings (string is already a pointer itself) and the only reason AWS SDK use them is because the types are generated from XML. Just use strings and empty value means "nothing". You will have to convert them, we have a package `ptr` with helper functions for that.